### PR TITLE
Shifting colors: Use `shift-color` instead of `shade/tint-color`

### LINKED ISSUE #32949: v5: shift-color() not working correctly with dark default link color?
If I want a dark color as my default link color (e.g. `#111111`), I would want its hover state to lighten the color instead of darkening it. It would appear that v5 accounts for this in its variables file, using the `shift-color()` function. 
```
$link-color: $primary !default;
$link-decoration: underline !default;
$link-shade-percentage: 20% !default;
$link-hover-color: shift-color($link-color, $link-shade-percentage) !default;
$link-hover-decoration: null !default;
```
According to `shift-color()`’s [comment](https://github.com/twbs/bootstrap/blob/v5.0.0-beta1/scss/_functions.scss#L164) , it should _“Shade the color if the weight is positive, else tint it_”. This sounds perfect and is exactly what I hope for. But if I pass a negative percentage like `-20%` to `$link-shade-percentage` in hopes of tinting it, I get a Sass compilation error: `$weight: Expected -20% to be within 0% and 100%`. 

This aligns with that Sass says, that [`mix()` only accepts 0% - 100% weight](https://sass-lang.com/documentation/modules/color#mix), but I don't know. Am I doing this wrong? Thanks for your work getting v5 out!

### PR DESCRIPTION
**Pull request by louismaximepiton**
Should fix #32949.

### Solution

Use `shift-color` instead of `shade-color` or `tint-color`. Adjust the Sass variables.

### Checks

- Couldn't spot any visual change on doc

### Need feedback

Do every change need to occur or only the one in the issue ?
Might have some breaking changes on some variables. Should some variables be renamed ?

### CODE REVIEW

File: scss/_buttons.scss [hunk 1]
```diff
@@ -119,19 +119,19 @@
       @include button-variant(
         $value,
         $value,
-        $hover-background: shade-color($value, $btn-hover-bg-shade-amount),
-        $hover-border: shade-color($value, $btn-hover-border-shade-amount),
-        $active-background: shade-color($value, $btn-active-bg-shade-amount),
-        $active-border: shade-color($value, $btn-active-border-shade-amount)
+        $hover-background: shift-color($value, $btn-hover-bg-shade-amount),
```
Comment (louismaximepiton) [inline scss/_buttons.scss:122]: These changes weren't asked but seems more consistent to me. Might bring some breaking changes.

File: scss/_variables.scss [hunk 2]
```diff
@@ -780,13 +780,13 @@ $btn-border-radius-lg:        $border-radius-lg !default;
 $btn-transition:              color .15s ease-in-out, background-color .15s ease-in-out, border-color .15s ease-in-out, box-shadow .15s ease-in-out !default;
 
 $btn-hover-bg-shade-amount:       15% !default;
-$btn-hover-bg-tint-amount:        15% !default;
+$btn-hover-bg-tint-amount:        -15% !default;
```
Comment (louismaximepiton) [inline scss/_variables.scss:783]: Should we keep these changes ?

**Comment by mdo** [comment]
I _think_ these might be breaking changes. Changing the core color function, and the values we use to set them (as in moving from positive to negative values), would be the problem. This probably has to be a v6 consideration—thoughts @ffoodd @julien-deramond?

**Comment by louismaximepiton** [comment]
Closing it for now, to clear the PR numbers.

**Pull request closed without merging**